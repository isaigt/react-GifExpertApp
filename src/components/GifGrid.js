import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria }) => {
    const { data:images, loading } = useFetchGif( categoria );

    return (
        <>
            <h3 className="card animate__animated animate__fadeInDown">{ categoria }</h3>
            { loading && <h4 className="animate__animated animate__rotateInDownLeft">Cargando...</h4> }
            <div className="card-grid">
                    {
                        images.map( imagen => (
                            <GifGridItem key={ imagen.id } {...imagen} />
                        ))
                    }
            </div>
        </>
    );
}
