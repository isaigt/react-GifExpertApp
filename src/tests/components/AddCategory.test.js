import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas del componente <AddCategory />', () => {
    const setCategorias = jest.fn();
    let wrapper = shallow( <AddCategory setCategorias={ setCategorias } /> ); 

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategorias={ setCategorias } /> );
    });

    test('se debe de mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('se debe actualizar el input al ocurrir el evento onChange', () => {
        const inp = wrapper.find('input');
        const value = "Hola mundo";

        inp.simulate('change', { target: { value } });
        expect( wrapper.find('p').text() ).toBe( value );
    });

    test('NO se debe de ejecutar la funcion setCategorias', () => {
        const frm =wrapper.find('form');
        frm.simulate('submit', { preventDefault(){} });
        expect( setCategorias ).not.toHaveBeenCalled();
    });

    test('debe de llamar el setCategorias y limpiar la caja de texto', () => {
        const valor = 'Hola Mundo';
        const inp = wrapper.find('input');
        inp.simulate( 'change', { target: { value: valor } } );
        wrapper.find('form').simulate( 'submit', { preventDefault(){} });

        expect( setCategorias ).toHaveBeenCalledWith( expect.any(Function) );
        expect( setCategorias ).toHaveBeenCalled();
        expect( inp.find('input').prop('value') ).toBe( '' );
    });
});
