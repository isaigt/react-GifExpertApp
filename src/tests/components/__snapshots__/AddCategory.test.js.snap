// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <AddCategory /> se debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
